src/Host/Client/Impl/rtvs/R/overrides.R

```diff
@@ -54,18 +54,29 @@ show_file <- function(files, header, title, delete.file) {
     }
 }
 
-
 # Override for edit(...). Opens file in VS editor and blocks until the file is closed.
-edit_file <- function(name, file, title) {
-    if(!is.null(name) && is.function(name)) {
+edit_file <- function(name = NULL, file = NULL, title = NULL) {
+    source <- NULL;
+    if(is.function(name)) {
         source <- send_request_and_get_response("?EditFile", paste0(deparse(name), collapse = '\n'), NULL);
-        parsed <- try(eval.parent(parse(text = source)));
-        return(parsed);
-    } else if(!is.null(file)) {
-        send_request_and_get_response("?EditFile", NULL, file);
-    } else {
-        utils:::edit(name, file, title);
+    } else if(!is.null(file) && is.character(file) && file != "") {
+        source <- send_request_and_get_response("?EditFile", NULL, file);
+    }
+    if(!is.null(source)) {
+        result <- try(eval.parent(parse(text = source)));
+        return(result);
+    }
+    editor <- getOption("externalEditor");
```

Comment:
Yes. It does call use back on editor(123). We are overriding editor and need to pass something to External2. It does not accepts nulls or missing arguments. I don't want to blindly pass 'notepad', perhaps user will want to provide something different via externalEditor option.

---
In reply to: [108814244](https://github.com/Microsoft/RTVS/pull/3362#discussion_r108814244) [](ancestors = 108814244)